### PR TITLE
 	Bug 1215617 - Filtering interface for perfherder compare

### PR DESCRIPTION
**Pull request by wlach**
Can now filter across test name and platform, as well as only
show significant changes and/or results for which we have confidence.

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1082)

### CODE REVIEW

File: ui/js/perfapp.js [hunk 1]
```diff
@@ -12,11 +12,11 @@ perf.config(function($compileProvider, $stateProvider, $urlRouterProvider) {
         controller: 'GraphsCtrl'
     }).state('compare', {
         templateUrl: 'partials/perf/comparectrl.html',
-        url: '/compare?originalProject&originalRevision&newProject&newRevision&hideMinorChanges&showExcludedPlatforms',
+        url: '/compare?originalProject&originalRevision&newProject&newRevision&hideMinorChanges&showExcludedPlatforms&filterTest&filterPlatform&showOnlyImportant&showOnlyConfident',
         controller: 'CompareResultsCtrl'
     }).state('comparesubtest', {
         templateUrl: 'partials/perf/comparesubtestctrl.html',
-        url: '/comparesubtest?originalProject&originalRevision&newProject&newRevision&originalSignature&newSignature&hideMinorChanges',
+        url: '/comparesubtest?originalProject&originalRevision&newProject&newRevision&originalSignature&newSignature',
```
Comment (jmaher) [inline ui/js/perfapp.js]: do we need to consider for the subtests: showOnlyImportant&showOnlyConfident

File: ui/js/directives/perf/compare.js [hunk 2]
```diff
@@ -8,10 +8,38 @@ treeherder.directive(
...
+                function filter(item, matchText) {
+                    return item.toLowerCase().indexOf(matchText.toLowerCase()) > (-1);
+                }
+                scope.filterTest = function(item) {
+                    return filter(item, scope.testFilter);
+                };
+                scope.filterPlatform = function(result) {
+                    return filter(result.name, scope.platformFilter) &&
+                        (!scope.showOnlyImportant || result.isMeaningful) &&
+                        (!scope.showOnlyConfident || result.isConfident);
```
Comment (jmaher) [inline ui/js/directives/perf/compare.js]: oh, this is a lot of logic!

File: ui/js/controllers/perf/compare.js [hunk 3]
```diff
@@ -238,9 +252,16 @@ perf.controller('CompareResultsCtrl', [
             }
 
             // we are excluding macosx 10.10 by default for now, see: https://bugzilla.mozilla.org/show_bug.cgi?id=1201615
-            $scope.excludedPlatforms = Boolean($stateParams.showExcludedPlatforms) ?
+            $scope.excludedPlatforms = Boolean(parseInt($stateParams.showExcludedPlatforms)) ?
                 [] : [ 'osx-10-10' ];
```
Comment (jmaher) [inline ui/js/controllers/perf/compare.js]: can you add a checkbox for excluded platforms?

File: ui/js/perfapp.js (hunk 1 above)
Comment (wlach) [inline ui/js/perfapp.js]: I thought about this a bit. I don't think so: subtest comparisons tend to be a lot shorter and you probably want the additional detail in all cases.

Even if we did decide it was extremely desirable, we can always do it in a followup.

**Comment by wlach** [comment]
Merged in 7f7f54f91cff381d2540cda2c5c67323016e932c

**Pull request closed without merging**